aten/src/ATen/core/Type.h

```diff
@@ -29,10 +29,13 @@
 #endif
 #endif
 
+namespace c10 {
+struct Allocator;
```

Comment:
Given that this includes Allocator.h, I don't think there's any point forward declaring it...